tests/_util/commands_tests.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+import unittest
+
+from google.datalab.utils.commands import CommandParser
+
+
+class TestCases(unittest.TestCase):
+
+  def test_subcommand_line(self):
+
+    parser = CommandParser(
+        prog='%test_subcommand_line',
+        description='test_subcommand_line description')
+
+    subcommand1 = parser.subcommand('subcommand1', help='subcommand1 help')
+    subcommand1.add_argument('--string1', help='string1 help.')
+    subcommand1.add_argument('--flag1', action='store_true', default=False,
+                             help='flag1 help.')
+
+    args, cell = parser.parse('subcommand1 --string1 value1 --flag1', None)
+    self.assertEqual(vars(args), {'string1': 'value1', 'flag1': True})
+    self.assertIsNone(cell)
+
+    args, cell = parser.parse('subcommand1 --string1 value1', 'string3: value3')
+    self.assertEqual(vars(args), {'string1': 'value1', 'flag1': False})
+    self.assertEqual(cell.strip(), 'string3: value3')
+
+    args, cell = parser.parse('subcommand1', 'string1: value1\nflag1: true')
+    self.assertEqual(vars(args), {'string1': 'value1', 'flag1': True})
+    self.assertIsNone(cell)
+
+    parser = CommandParser(
```

Comment:
the subcommand test can be put into a new test function. So test errors are easier to debug.